App.js

```diff
@@ -10,42 +10,35 @@ import {
 } from 'react-native'
 import { NavigationContainer } from '@react-navigation/native'
 import Tabs from './src/components/Tabs'
-import { useGetWeather } from './src/hooks/useGetWeather'
 import ErrorItem from './src/components/ErrorItem'
-
-export const WeatherContext = createContext()
+import {
+    WeatherContext,
+    WeatherProvider
+} from './src/components/WeatherDataContext'
 
 const App = () => {
-    const [loading, error, weather] = useGetWeather()
-    const [refreshing, setRefreshing] = useState(false)
-    console.log(JSON.stringify(weather.city, null, 6) + 'More words')
-
-    const onRefresh = useCallback(() => {
-        setRefreshing(true)
-        setTimeout(() => {
-            setRefreshing(false)
-        }, 2000)
-    }, [])
+    const { loading, error, weather } = useContext(WeatherContext)
```

Comment:
remember that the useContext gets **the nearest provider for that context**

your only WeatherContext provider is in your child, so it aint gonna find one, and will fail.